.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,54 @@
+name: Benchmark
+
+on:
+  push:
+    branchs: [ master ]
+  pull_request:
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['10.x', '12.x', '14.x']
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Cache NPM dependencies
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.OS }}-npm-cache
+          restore-keys: |
+            ${{ runner.OS }}-npm-cache
+      - run: mkdir .tmp-hexo-theme-unit-test
+      - name: Setting up a dummy hexo site with 500 posts
+        uses: actions/checkout@v2
+        with:
+          repository: hexojs/hexo-theme-unit-test
+          path: .tmp-hexo-theme-unit-test
+      - name: Install hexo-theme-landscape
+        uses: actions/checkout@v2
+        with:
+          repository: hexojs/hexo-theme-landscape
+          path: .tmp-hexo-theme-unit-test/themes/landscape
+      - name: Install Dependencies
+        run: npm install --silent
+        working-directory: .tmp-hexo-theme-unit-test
+      - name: Replace node_modules/hexo
+        run: |
+          rm -rf node_modules/hexo
+          ln -sf $GITHUB_WORKSPACE node_modules/hexo
+        working-directory: .tmp-hexo-theme-unit-test
```

Comment:
IMHO I prefer running those setup inside `benchmark.js`, so that we could run benchmark on our local machine as well (simply `node benchmark.js` instead of manually setup the benchmark environment).

Reply:
OK:) If you can use git command in github actions runner, that is better. I'll give it a try.